File: frontend/public/logos/create_logos.py
from PIL import Image, ImageDraw, ImageFont

def create_logo(filename, bg_color, text_color, letter, name):
    # Create image
    img = Image.new('RGB', (200, 200), color=bg_color)
    draw = ImageDraw.Draw(img)
    
    try:
        # Try to use a nice font
        font_large = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 80)
        font_small = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 20)
    except:
        # Fallback to default font
        font_large = ImageFont.load_default()
        font_small = ImageFont.load_default()
    
    # Draw letter
    bbox = draw.textbbox((0, 0), letter, font=font_large)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    x = (200 - text_width) // 2
    y = 60
    draw.text((x, y), letter, fill=text_color, font=font_large)
    
    # Draw name
    bbox_small = draw.textbbox((0, 0), name, font=font_small)
    text_width_small = bbox_small[2] - bbox_small[0]
    x_small = (200 - text_width_small) // 2
    y_small = 140
    draw.text((x_small, y_small), name, fill=text_color, font=font_small)
    
    # Save
    img.save(filename)
    print(f"Created {filename}")

# Create logos
create_logo('uber.png', '#000000', '#FFFFFF', 'U', 'UBER')
create_logo('rapido.png', '#FFC107', '#000000', 'R', 'RAPIDO')
create_logo('namma_yatri.png', '#7C3AED', '#FFFFFF', 'NY', 'NAMMA YATRI')

print("All logos created successfully!")

File: frontend/public/logos/create_logos.js
const fs = require('fs');
const { createCanvas } = require('canvas');

function createLogo(filename, bgColor, textColor, letter, name) {
  const canvas = createCanvas(200, 200);
  const ctx = canvas.getContext('2d');
  
  // Fill background
  ctx.fillStyle = bgColor;
  ctx.fillRect(0, 0, 200, 200);
  
  // Draw letter
  ctx.fillStyle = textColor;
  ctx.font = 'bold 80px Arial';
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText(letter, 100, 80);
  
  // Draw name
  ctx.font = 'bold 20px Arial';
  ctx.fillText(name, 100, 140);
  
  // Save
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(filename, buffer);
  console.log(`Created ${filename}`);
}

// Create logos
createLogo('uber.png', '#000000', '#FFFFFF', 'U', 'UBER');
createLogo('rapido.png', '#FFC107', '#000000', 'R', 'RAPIDO');
createLogo('namma_yatri.png', '#7C3AED', '#FFFFFF', 'NY', 'NAMMA YATRI');

console.log('All logos created successfully!');
